### ISSUE TITLE
Cypress stuck in infinite redirect loop

### ISSUE THREAD

**Issue by esbarila** [cypress-io/cypress#27185]
EDIT: replicate this bug in a simpler website: cy.visit() to saucedemo.com using cypres open to run the test.

### Current behavior

1. cy.visit timeouts even when page is fully visible in the runner. URL is http://tutorialspoint.com/html/html_iframes.htm

> Your page did not fire its load event within 60000ms

This timeout issue was reported by several people in stackoverflow without a solution.
https://stackoverflow.com/questions/76186375/how-to-resolve-or-ignore-the-your-page-did-not-fire-its-load-event-error
https://stackoverflow.com/questions/76591451/visiting-tutorialspoint-with-cypress-keeps-waiting-for-the-page-to-load

Some people report it in comments of other post to StackOverflow, and decide [to quit Cypress altogether as a testing framework](https://stackoverflow.com/questions/71984376/your-page-did-not-fire-its-load-event-within-60000ms-only-on-github-actions?noredirect=1#comment135052347_71984376). 
[Also here](https://stackoverflow.com/questions/76186375/how-to-resolve-or-ignore-the-your-page-did-not-fire-its-load-event-error?noredirect=1#comment135057078_76186375)

1. ![Screenshot from 2023-07-02 13-05-19](https://github.com/cypress-io/cypress/assets/47305731/da50fe70-3477-4bd7-bf7a-27267a894cc5)

### Desired behavior

`cy.visit('/http://tutorialspoint.com/html/html_iframes.htm');`
or 
`baseURL: "http://tutorialspoint.com/html/html_iframes.htm"` with `cy.visit('/')`

Should work at visiting the page, rendering it as experienced by an end user on chrome and allow me to continue to test the website.

### Test code to reproduce

```
describe('Cypress interaction with Tutorialspoint', () => {
    beforeEach(() => {
        cy.visit('http://tutorialspoint.com/html/html_iframes.htm'); //FAILS HERE!!!!!
        // cy.visit('/'); in case is set at baseURL
    })

    it('Performs tasks and logs results', () =>{
        cy.contains('HTML - Iframes'); //THIS ASSERTION IS MEANINGLESS. READ THE DESCRIPTION.
    })
})
```

### Cypress Version

12.16.0

### Node version

v18.16.1

### Operating System

Ubuntu 22.04.2 LTS x86_64

### Debug Logs

```shell
When cy visit, url: CypressError: Timed out after waiting `60000ms` for your remote page to load.
Your page did not fire its `load` event within `60000ms`.
You can try increasing the `pageLoadTimeout` value in `cypress.config.js` to wait longer.
Browsers will not fire the `load` event until all stylesheets and scripts are done downloading.
When this `load` event occurs, Cypress will continue running commands.
```

### Other

selecting https in the URL works the same way as withouth

**Comment by MikeMcC399** [comment]
@esbarila 

You would normally want to set

`baseURL: 'http://tutorialspoint.com/'`
then use
`cy.visit(/html/html_iframes.htm)`

There is no `cy.visit()` timeout in this case, however you would then need to deal with accepting cookies.

Note that Cypress has restrictions about testing the content of `<iframes>` (see https://docs.cypress.io/faq/questions/using-cypress-faq#How-do-I-test-elements-inside-an-iframe).

You will find some of these aspects dealt with in various Cypress training courses.

Also if you need to connect to the Cypress technical community on Discord you can find them here:

[![Discord chat](https://img.shields.io/badge/chat-on%20Discord-brightgreen)](https://on.cypress.io/discord)

Discord is used by experienced and beginner users of Cypress.

**Comment by esbarila** [comment]
@MikeMcC399 Hi Mike. I've already gone that route and still get "Your page did not fire its load event within 60000ms."
![image](https://github.com/cypress-io/cypress/assets/47305731/2be2cdd8-41e0-4c98-a1db-a1b4d65bf638)

**Comment by MikeMcC399** [comment]
@esbarila 

You may run into other issues, like uncaught exceptions, however when I tried it out, the page loaded:

![image](https://github.com/cypress-io/cypress/assets/66998419/90b4324c-2ae1-4de1-92f3-e50545cae8f8)

Are you testing this page as a learning exercise or is it part of your job to test the page?

**Comment by esbarila** [comment]
@MikeMcC399 its part of a job posting challenge. I guess the iframe part is, although I haven't get there yet with Cypress. 

This is so weird, I tried it like you using Electron but I keep getting the same. Also I've never saw that consent popup before.

BTW: I've noticed opening this website in incognito makes it so that it never ends loading (wheel keeps spinning in the tab). Regular chrome session doesn't do that.

Anyway here's the code, do you see anything clearly wrong with it?
https://github.com/esbarila/AutoFi-Zygmus-challenge/tree/main/cypress

**Comment by MikeMcC399** [comment]
@esbarila 

In your https://github.com/esbarila/AutoFi-Zygmus-challenge/tree/main/cypress repo you try to `.click()`, however the cookie pop-up prevents that.

Cypress issues are intended to report bugs and I don't think that you have found any bug so far.

If you want to get advice from other Cypress users then you can use Discord. You could also ask them if they think that Cypress is the right tool to complete your challenge.

BTW: For a production repo I would add

```text
cypress/downloads
cypress/screenshots
cypress/videos
```

to `.gitignore` as normally these temporary files would not be committed. See https://docs.cypress.io/guides/core-concepts/writing-and-organizing-tests#Asset-Files

**Comment by esbarila** [comment]
@MikeMcC399 Understand my point: my test never starts for some reason. Have you tried matching my context?

**Comment by MikeMcC399** [comment]
@esbarila 

Your test as you provided it in the repo https://github.com/esbarila/AutoFi-Zygmus-challenge/tree/main/cypress fails with the error logged below:

```
$ npx cypress run


DevTools listening on ws://127.0.0.1:49871/devtools/browser/94842fd4-6770-4d37-aabd-d4c3ba983a2e

====================================================================================================

  (Run Starting)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Cypress:        12.16.0                                                                        │
  │ Browser:        Electron 106 (headless)                                                        │
  │ Node Version:   v18.16.1 (C:\Program Files\nodejs\node.exe)                                    │
  │ Specs:          1 found (tutorialsPoint.cy.js)                                                 │
  │ Searched:       cypress/e2e/**/*.cy.{js,jsx,ts,tsx}                                            │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


────────────────────────────────────────────────────────────────────────────────────────────────────

  Running:  tutorialsPoint.cy.js                                                            (1 of 1)


  Cypress interaction with Tutorialspoint
    1) Performs tasks and logs results


  0 passing (17s)
  1 failing

  1) Cypress interaction with Tutorialspoint
       Performs tasks and logs results:
     CypressError: Timed out retrying after 10050ms: `cy.click()` failed because this element:

`<a href="/html/html_iframes.htm">HTML - ...</a>`

is being covered by another element:

`<div class="fc-dialog-overlay"></div>`

Fix this problem, or use {force: true} to disable error checking.

https://on.cypress.io/element-cannot-be-interacted-with
      at ensureIsDescendent (https://www.tutorialspoint.com/__cypress/runner/cypress_runner.js:139711:76)
      at ensureDescendents (https://www.tutorialspoint.com/__cypress/runner/cypress_runner.js:139762:5)
      at ensureDescendentsAndScroll (https://www.tutorialspoint.com/__cypress/runner/cypress_runner.js:139768:14)
      at ensureElIsNotCovered (https://www.tutorialspoint.com/__cypress/runner/cypress_runner.js:139899:5)
      at runAllChecks (https://www.tutorialspoint.com/__cypress/runner/cypress_runner.js:140079:52)
      at retryActionability (https://www.tutorialspoint.com/__cypress/runner/cypress_runner.js:140116:16)
      at tryCatcher (https://www.tutorialspoint.com/__cypress/runner/cypress_runner.js:18744:23)
      at Promise.attempt.Promise.try (https://www.tutorialspoint.com/__cypress/runner/cypress_runner.js:16018:29)
      at whenStable (https://www.tutorialspoint.com/__cypress/runner/cypress_runner.js:156384:65)
      at <unknown> (https://www.tutorialspoint.com/__cypress/runner/cypress_runner.js:155825:14)
      at tryCatcher (https://www.tutorialspoint.com/__cypress/runner/cypress_runner.js:18744:23)
      at Promise._settlePromiseFromHandler (https://www.tutorialspoint.com/__cypress/runner/cypress_runner.js:16679:31)
      at Promise._settlePromise (https://www.tutorialspoint.com/__cypress/runner/cypress_runner.js:16736:18)
      at Promise._settlePromise0 (https://www.tutorialspoint.com/__cypress/runner/cypress_runner.js:16781:10)
      at Promise._settlePromises (https://www.tutorialspoint.com/__cypress/runner/cypress_runner.js:16861:18)
      at Promise._fulfill (https://www.tutorialspoint.com/__cypress/runner/cypress_runner.js:16805:18)
      at <unknown> (https://www.tutorialspoint.com/__cypress/runner/cypress_runner.js:18419:46)
  From Your Spec Code:
      at Context.eval (webpack:///./cypress/e2e/tutorialsPoint.cy.js:7:38)




  (Results)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Tests:        1                                                                                │
  │ Passing:      0                                                                                │
  │ Failing:      1                                                                                │
  │ Pending:      0                                                                                │
  │ Skipped:      0                                                                                │
  │ Screenshots:  1                                                                                │
  │ Video:        true                                                                             │
  │ Duration:     16 seconds                                                                       │
  │ Spec Ran:     tutorialsPoint.cy.js                                                             │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


  (Screenshots)

  -  D:\github2\AutoFi-Zygmus-challenge\cypress\screenshots\tutorialsPoint.cy.js\Cypr     (1280x720)
     ess interaction with Tutorialspoint -- Performs tasks and logs results (failed).
     png


  (Video)

  -  Started compressing: Compressing to 32 CRF
  -  Finished compressing: 1 second

  -  Video output: D:\github2\AutoFi-Zygmus-challenge\cypress\videos\tutorialsPoint.cy.js.mp4


====================================================================================================

  (Run Finished)


       Spec                                              Tests  Passing  Failing  Pending  Skipped
  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ ✖  tutorialsPoint.cy.js                     00:16        1        -        1        -        - │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘
    ✖  1 of 1 failed (100%)                     00:16        1        -        1        -        -
```

I corrected the problem by changing the line to include `.click({ force: true })`

`        cy.contains('HTML - Iframes').click({ force: true });`

and the test then ran using

`$ npx cypress run`

**Comment by esbarila** [comment]
@MikeMcC399 Again, that is not my output and that test is meaningless.

My issue is cy.visit() failing. I think I made that clear.

> Your page did not fire its load event within 60000ms

I just don't know why its working for you and not for me.

**Comment by MikeMcC399** [comment]
@esbarila 

I'm sorry that I cannot help you further. After I corrected the test error, the test ran successfully. `cy.visit()` worked. I do not know why it does not work for you.

![image](https://github.com/cypress-io/cypress/assets/66998419/b4dd11e6-18e1-42cd-8b95-9e945900d550)

**Comment by esbarila** [comment]
I already solved it using Testcafe, but it would be useful to know which versions of Cypress Node you used. Also which OS. @MikeMcC399

**Comment by MikeMcC399** [comment]
@esbarila 

> but it would be useful to know which versions of Cypress Node you used. Also which OS

You can see the versions in the log I posted.
The OS is Windows 11.

I didn't try it on Ubuntu. I don't expect it would have made any difference.
  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Cypress:        12.16.0                                                                        │
  │ Browser:        Electron 106 (headless)                                                        │
  │ Node Version:   v18.16.1 (C:\Program Files\nodejs\node.exe)                                    │
  │ Specs:          1 found (tutorialsPoint.cy.js)                                                 │
  │ Searched:       cypress/e2e/**/*.cy.{js,jsx,ts,tsx}                                            │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘

In the past the Cypress team has commented that they don't support testing on sites you don't control (see https://docs.cypress.io/guides/references/best-practices#Visiting-External-Sites).

If you have solved your challenge, then I suggest to close this issue.

**Comment by nagash77** [comment]
Hi @esbarila looking at your test repo I see you have not added the `{ force: true }` argument to your test.  If you add that into the test are you seeing the test work as expected?  Have you also tried taking out the beforeEach call and simplifying to verify the test works in the most straight forward scenario?

**Comment by esbarila** [comment]
> Hi @esbarila looking at your test repo I see you have not added the `{ force: true }` argument to your test. If you add that into the test are you seeing the test work as expected? Have you also tried taking out the beforeEach call and simplifying to verify the test works in the most straight forward scenario?

As discussed earlier, that assertion is meaningless. Im going to edit it out. I can't get past CY.VISIT() for some reason and I wan't to know why. Already tried in 2 different computers.

**Comment by ghost** [comment]
Hi @esbarila I too getting same error while I tried to run through CI. If you find any solution kindly respond to this issue.

Node v14.19.0

cypress v12.16.0

CypressError: Timed out after waiting `120000ms` for your remote page to load.

Your page did not fire its `load` event within `120000ms`.

You can try increasing the `pageLoadTimeout` value in `cypress.config.js` to wait longer.

Browsers will not fire the `load` event until all stylesheets and scripts are done downloading.

When this `load` event occurs, Cypress will continue running commands.

**Comment by dsolisp** [comment]
> @esbarila
> 
> > but it would be useful to know which versions of Cypress Node you used. Also which OS
> 
> You can see the versions in the log I posted. The OS is Windows 11.
> 
> I didn't try it on Ubuntu. I don't expect it would have made any difference. ┌────────────────────────────────────────────────────────────────────────────────────────────────┐ │ Cypress: 12.16.0 │ │ Browser: Electron 106 (headless) │ │ Node Version: v18.16.1 (C:\Program Files\nodejs\node.exe) │ │ Specs: 1 found (tutorialsPoint.cy.js) │ │ Searched: cypress/e2e/**/*.cy.{js,jsx,ts,tsx} │ └────────────────────────────────────────────────────────────────────────────────────────────────┘
> 
> In the past the Cypress team has commented that they don't support testing on sites you don't control (see https://docs.cypress.io/guides/references/best-practices#Visiting-External-Sites).
> 
> If you have solved your challenge, then I suggest to close this issue.

the issue is very reproducible, as it says

there is no much code to add, just do a cy.visit() and for some reason when testing some pages it happens randomly.

my students are having lots of problems doing the visit command with https://www.saucedemo.com/ page

Im not sure why you guys are focusing on the whole test class, is just a simple cy.visit() failing with 

When cy visit, url: CypressError: Timed out after waiting `60000ms` for your remote page to load.
Your page did not fire its `load` event within `60000ms`.
You can try increasing the `pageLoadTimeout` value in `cypress.config.js` to wait longer.
Browsers will not fire the `load` event until all stylesheets and scripts are done downloading.
When this `load` event occurs, Cypress will continue running commands.

**Comment by MikeMcC399** [comment]
@dsolisp 

> the issue is very reproducible, as it says
>
> there is no much code to add, just do a cy.visit() and for some reason when testing some pages it happens randomly.
>
> my students are having lots of problems doing the visit command with https://www.saucedemo.com/ page

I tried the following and it loaded with no error:

```
describe('template spec', () => {
  it('passes', () => {
    cy.visit('https://www.saucedemo.com/')
  })
})
```

If you want to have your problem specifically looked at then I suggest that you open a new issue and provide the details that the form requests. Although your error message is the same as the one from the original post here, the root cause may be different, so it will get confusing to try to analyse two different set ups in one issue.

You can also contact your technical community peers on 
[![Discord chat](https://img.shields.io/badge/chat-on%20Discord-brightgreen)](https://on.cypress.io/discord).

**Comment by MikeMcC399** [comment]
@dsolisp 
I have now seen this issue happening for https://www.saucedemo.com/ so I will see what I can find out.

**Comment by MikeMcC399** [comment]
@dsolisp 
I found the following, testing on Windows 11

1. `npx cypress run --e2e --browser electron`
runs without issue
2. `npx cypress open --e2e --browser electron`
then select and run test, displays the contents of https://www.saucedemo.com (Swag Labs) and after 60 secs displays the error
"Your page did not fire its load event within 60000ms"
3. In Google Chrome F12 shows Load: 575 ms

Edit: I cloned repo https://github.com/saucelabs/sample-app-web, which is the source repo for the deployment to GitHub Pages on https://www.saucedemo.com. I ran it locally with Cypress `12.17.0` and it did not show any timeout error with `baseUrl` set to `http://localhost:3000/` and calling `cy.visit('/')`.

The repo [README](https://github.com/saucelabs/sample-app-web/blob/main/README.md) says that it is dependent on Node.js `14` which entered [end-of-life](https://github.com/nodejs/release#release-schedule) on April 30, 2023, meaning that the source code is out-of-date.

Using a locally run copy instead of testing against https://www.saucedemo.com may be a workaround for the issue.

**I would encourage you to open your own separate issue for https://www.saucedemo.com/ if the workaround does not meet your needs.**

You may also want to contact https://github.com/saucelabs/sample-app-web if you are relying on that code.

**Comment by MikeMcC399** [comment]
@esbarila 

After the page http://tutorialspoint.com/html/html_iframes.htm displays in the Cypress app and the clock stops counting, it is continually contacting tracking sites  . When I left it like this using the Electron browser it said "We detected that the Chromium Renderer process just crashed."

When I tried it with Google Chrome it eventually crashed with "Error code: SIGTRAP". It took longer for Chrome to fail.

This was whilst it was waiting for user input regarding the permissions to collect data (cookies). So it seems strange it should be contacting all these sites **before** any consent for data collection has been given.

**Comment by MikeMcC399** [comment]
@esbarila 

The problem with http://tutorialspoint.com/html/ occurs in the Cypress app (started with `cypress open`). After the page has displayed, it continues indefinitely with network activity and the Cypress app records all this activity as part of its [Time Traveling](https://docs.cypress.io/guides/core-concepts/cypress-app#Time-traveling) capability. It is also recording a video by default in parallel. At some point the browser memory is exhausted because of the recorded data. Depending on which browser is used and other environmental conditions (operating system, processor, memory available, etc.), it may first report that the page has not loaded "Your page did not fire its load event".

The repro repo https://github.com/esbarila/AutoFi-Zygmus-challenge `main` branch does not show this issue. It shows another issue. Deleting
https://github.com/esbarila/AutoFi-Zygmus-challenge/blob/7052f16123cfa48dc923aa91d1adc16496e4d1ac/cypress/e2e/tutorialsPoint.cy.js#L7
`        cy.contains('HTML - Iframes').click();`
does however allow a problem with `cy.visit()` to be shown.

`npx cypress run`
does not fail. In other words `cy.visit()` is not causing an error.

`npx cypress open --e2e` followed by running `tutorialsPoint.cy.js` loads the page and continues network activity. After some time, Electron crashes with "We detected that the Chromium Renderer process just crashed." . Chrome also reported a crash. Firefox simply hung up and became unresponsive.

To avoid recording the on-going network activity, which leads to browser failure, add the following to `cypress.config. js`:

```json
numTestsKeptInMemory: 0,
video: false,
```

see [Global config options](https://docs.cypress.io/guides/references/configuration#Global)
see [Videos config options](https://docs.cypress.io/guides/references/configuration#Videos)

Please check if making this change also stops the error "Your page did not fire its load event" occurring.

**Comment by esbarila** [comment]
@MikeMcC399 On my end, `npx cypress run` keeps timing out at cy.visit() with the test running headed or on the background.

I also tried with that config and the same thing happens.

**Comment by MikeMcC399** [comment]
@esbarila 

> On my end, `npx cypress run` keeps timing out at cy.visit() with the test running headed or on the background.

I could only get a timeout to show if I used `npx cypress open` and used Google Chrome's throttling simulation.

"You can try increasing the pageLoadTimeout value in cypress.config.js to wait longer."

So for instance you could add a higher value like 120 seconds:

```json
pageLoadTimeout: 120000,
```

see [Timeout config options](https://docs.cypress.io/guides/references/configuration#Timeouts)

If that doesn't work, perhaps @lmiller1990 can help.

**Comment by esbarila** [comment]
im not waiting more than 1 minute to load the page, there's something wrong with cypress. The page loads completely in my browser in 1 second.

**Comment by lmiller1990** [comment]
> I could only get a timeout to show if I used npx cypress open and used Google Chrome's throttling simulation.

I wonder if it is a resource issue.  @esbarila are you able to share information about your machine? RAM, CPU etc.


https://github.com/cypress-io/cypress/assets/19196536/b2bf4bc7-5669-4dac-aa87-18e687acd2f4

The reason I suspect this is happening on this particular website is due to the gigantic amount of trackers and redirects happening. Cypress usually waits for the page to be loaded - even on my beefy macbook, it's pretty damn slow - 20 seconds to 1 minute in extreme cases:

![image](https://github.com/cypress-io/cypress/assets/19196536/3e645eb1-366a-4020-acfc-cf07ba65ab47)

Cypress also adds overhead - we have a proxy that all resources pass through, which facilitates things like `cy.intercept()` etc. This will make things slower in Cypress, but it isn't usually this significant.

Note that loaded from a user point of view (I can see the page) and from a test runner point of view (requests have all settled) are different things. I will poke around a bit more, but I u

**Comment by lmiller1990** [comment]
I diagnosed the problem. One of hundreds of JavaScript resources loaded by this website has an infinite redirect. Opening the website in a regular browser seems to be okay; the redirect still happens, but the browser recognises the loop and terminates it.

Here's the error in my console (not Cypress, just Chrome). Note ERR_TOO_MANY_REDIRECTS.

![image](https://github.com/cypress-io/cypress/assets/19196536/7b8afdfe-2285-4bcd-a42a-29b4f99aa1ba)

I did `cy.visit()` in my test to see what happens. My beefy macbook pro freezes up due to the infinite loop. This is as far as I got before the entire thing was unresponsive:

https://github.com/cypress-io/cypress/assets/19196536/37ac01c1-bde7-4d59-9e9f-cb5976364d6c

It's not a resources issue. Cypress is waiting for the redirect loop to stop. 

It's not clear _why_ the redirect loop isn't detected by Cypress and terminated. I thought we'd get that for free from the browser. I'm not sure how / if / when we can fix this. 

In the meantime, the best solution would be to find out why the website has an infinite redirect loop and fix that - it's definitely a bug in the website, although I agree Cypress really should provide a more actionable error (not sure how to detect this - it's basically the hauling problem. I'm guessing when you visit in a regular browser session, it'll bail after N redirects).

As for your situation right now -- I'm not really sure what else to recommend. The best response I can come up with is "the website under test has a bug that is leading to an infinite redirect caused by X resource (TODO: find out which one). The test automation tool, Cypress, is unable to detect this, resulting in the test hanging due to getting stuck in an infinite redirect loop. I'd recommending inspecting X resource to find out why it's cause the redirect".

________

As for @dsolisp - https://www.saucedemo.com/ is working great for me. I'm not able to reproduce the experience you are having. Can you share some more information about your system?

**Comment by MikeMcC399** [comment]
@lmiller1990 

After adding the following `blockHosts` array to `cypress.config.js` for http://tutorialspoint.com/html/, the calls were replaced with HTTP `503` responses and the test in the Cypress app ran through.

```
  blockHosts: [
    'www.gstatic.com',
    'bat.bing.com',
    'cdn.rawgit.com',
    'pbs.nextmillmedia.com',
    '*.adpushup.com',
    'match.deepintent.com',
    'secure.cdn.fastclick.net',
    '*.amazon-adsystem.com',
    'shop.pe',
    'googletagmanager.com'
  ],
```

I didn't see the site which you saw (`cm.g.doubleclick.net`?) showing `ERR_TOO_MANY_REDIRECTS`, however it ought to be possible to block this site as well if it is causing issues.

**Comment by MikeMcC399** [comment]
@charan030 

> I too getting same error while I tried to run through CI. If you find any solution kindly respond to this issue.

What is the URL of the site you are trying to test? It is not so likely that your root cause will be the same as the one found for http://tutorialspoint.com/html/html_iframes.htm.

**Comment by lmiller1990** [comment]
Blocking all the trackers in a testing environment is a great idea @MikeMcC399 - I doubt any production website wants to count robots as real users, it messes up analytics.

**Comment by Lola-I** [comment]
I used a whitelist with `cy.intercept()` to allow only page-relevant requests to pass. It also allows the test to run faster.
  
```js
const whitelist = ['tutorialspoint.com', 'bing.com']
cy.intercept('*', (req) => {
  if (!whitelist.some(wl => req.url.includes(wl))) {
    req.reply({})
  }
})
cy.visit('https://www.tutorialspoint.com/html/index.htm');
```

**Comment by MikeMcC399** [comment]
@esbarila 

Is there anything more to be done with this issue? It seems it could be closed now.

**Comment by esbarila** [comment]
> @esbarila 
> 
> Is there anything more to be done with this issue? It seems it could be closed now.

Let me see. I'll edit this comment later.

**Comment by esbarila** [comment]
> > @esbarila
> > > but it would be useful to know which versions of Cypress Node you used. Also which OS
> > 
> > 
> > You can see the versions in the log I posted. The OS is Windows 11.
> > I didn't try it on Ubuntu. I don't expect it would have made any difference. ┌────────────────────────────────────────────────────────────────────────────────────────────────┐ │ Cypress: 12.16.0 │ │ Browser: Electron 106 (headless) │ │ Node Version: v18.16.1 (C:\Program Files\nodejs\node.exe) │ │ Specs: 1 found (tutorialsPoint.cy.js) │ │ Searched: cypress/e2e/**/*.cy.{js,jsx,ts,tsx} │ └────────────────────────────────────────────────────────────────────────────────────────────────┘
> > In the past the Cypress team has commented that they don't support testing on sites you don't control (see https://docs.cypress.io/guides/references/best-practices#Visiting-External-Sites).
> > If you have solved your challenge, then I suggest to close this issue.
> 
> the issue is very reproducible, as it says
> 
> there is no much code to add, just do a cy.visit() and for some reason when testing some pages it happens randomly.
> 
> my students are having lots of problems doing the visit command with https://www.saucedemo.com/ page
> 
> Im not sure why you guys are focusing on the whole test class, is just a simple cy.visit() failing with
> 
> When cy visit, url: CypressError: Timed out after waiting `60000ms` for your remote page to load. Your page did not fire its `load` event within `60000ms`. You can try increasing the `pageLoadTimeout` value in `cypress.config.js` to wait longer. Browsers will not fire the `load` event until all stylesheets and scripts are done downloading. When this `load` event occurs, Cypress will continue running commands.

I'm having the same problem now trying to cy.visit() saucedemo.com too. I don't understand why.
NOTE: only happens if I go though npx cypress open, otherwise npx cypress run makes it work alright.

**Comment by MikeMcC399** [comment]
> I'm having the same problem now trying to cy.visit() saucedemo.com too. I don't understand why.

See my comments in https://github.com/cypress-io/cypress/issues/27185#issuecomment-1625614195 about `saucedemo`.

**Comment by esbarila** [comment]
@MikeMcC399 this is an issue with cypress. Comment did not help.

**Comment by MikeMcC399** [comment]
Apart from the details in the comment, the main points would be:

**I would encourage you to open your own separate issue for https://www.saucedemo.com/ if the workaround does not meet your needs.**

You may also want to contact https://github.com/saucelabs/sample-app-web if you are relying on that code.

**Comment by esbarila** [comment]
@MikeMcC399 
It seriously boggles my mind you'd think this to be an issue on the side of Saucelabs, when cypress is a testing framework. It should adapt to whatever website you use it on.

Furthermore, there are no weird network requests or anything in that website. Nothing fishy there. 

And the most glaring thing of all, it works when using cypress run, not when running from the cypress open suite. Both times in chrome.

**Comment by MikeMcC399** [comment]
@esbarila 

- I believe this was a misunderstanding, as I was suggesting that you open a new Cypress issue, which you have now done in https://github.com/cypress-io/cypress/issues/27501. This should allow to find any bug in Cypress accessing https://www.saucedemo.com/.  To continue in this issue would have been confusing, since this issue was about using Cypress to access the page http://tutorialspoint.com/html/html_iframes.htm which behaves differently compared to https://www.saucedemo.com/.

- The mention of the repo https://github.com/saucelabs/sample-app-web which generates the website https://www.saucedemo.com/ was because the instructions are based on Node.js `14` which recently passed into [end-of-life](https://github.com/nodejs/release#release-schedule)
![image](https://github.com/cypress-io/cypress/assets/66998419/f990871b-d41c-4628-ace7-e1ee0fc52e0b)

**Comment by mpoetzsch** [comment]
maybe it the same problem like here: https://github.com/cypress-io/cypress/issues/25891

Summarization of the ticket:

the issue arises only with newer versions of Chrome (it worked for me with an older version). The problem likely started with the introduction of https://blog.chromium.org/2023/08/towards-https-by-default.html.

Workaround:
change the URL from https to http in the address bar of chrome and reload the web specs runner. After that, the tests run smoothly.

or a more permanent workaround is here:
https://superuser.com/questions/565409/how-to-stop-an-automatic-redirect-from-http-to-https-in-chrome/881431
(the solution with the most upvotes)

**Comment by kasikcz** [comment]
Hello, so we are currently having similar issue to this when running our test in Azure pipelines sometimes tests stucks on cy.visit and instead of failing after timeout it stucks there for 15min+- if we add uncaught exception to return false it then continue with test and pass, if it runs without uncaught exception it fails after the long wait. 

I tried to simulate it locally and when i added blockedhost as described from @MikeMcC399 just the selection which i found in logs that we are using. Then local run start doing it too even worse way first test in spec passed normally and then 2nd is stuck for like 45min and in console is just regular "your page did not fire its `load` event within `60000ms`" 
![image](https://github.com/cypress-io/cypress/assets/16073211/dbdd4ad9-8fbc-4e16-b308-332d9d2b1a07)
What can be seen on scrn is that in left bottom corner is some waiting for region1.google-analytics.com
It's transient and not happening every time...

Currently using Cy 13.0.0 - tried even with 13.6.0
used browser - Electron (tried chrome same behavior)
